s3transfer/upload.py

```diff
@@ -287,8 +289,12 @@ def _get_upload_part_fileobj_with_full_size(self, fileobj, **kwargs):
         # source. So we must treat it as its own standalone file.
         return six.BytesIO(data), len(data)
 
-    def _get_put_object_fileobj(self, fileobj):
-        return fileobj
+    def _get_put_object_fileobj_with_full_size(self, transfer_future):
+        fileobj = transfer_future.meta.call_args.fileobj
+        # The current position needs to be taken into account when retrieving
+        # the full size of the file.
+        size = fileobj.tell() + transfer_future.meta.size
```

Comment:
Yeah so ``transfer_future.meta.size()`` is always equal to how many bytes we expect to transfer over the wire. So if we are given a file of size 10 and the file is at position 6 of the file. We expect to transfer 4 bytes and that is what the ``transfer_future.meta.size`` value should be.

With how the ReadFileChunk works, it requires the full file size to determine how many bytes to return with basically the calculation:
```py
def read(self, amt):
   amt_to_read = min(amt, self.full_size - self.current_position)
   self.underlying_fileobj.read(amt_to_read)
```
So if we do not account for the full size of the file (``expected_size_to_transfer (aka transfer_future.meta.size) + current_pos``) we will get incorrect numbers. So if we take the previous example, if the file is at position 6 and the actual total file size is 10 and instead of using 10 we use ``transfer_future.meta.size`` which is 4, the read file chunk will calculate ``self.full_size - self.current_position`` to be -1 which is wrong and will use -1 to make the read because it is the minimum. 

Does this make sense?